src/responder/common/responder_dp.c

```diff
@@ -327,3 +327,128 @@ sss_dp_get_account_recv(TALLOC_CTX *mem_ctx,
 
     return EOK;
 }
+
+struct sss_dp_resolver_get_state {
+    uint16_t dp_error;
+    uint32_t error;
+    const char *error_message;
+};
+
+static void sss_dp_resolver_get_done(struct tevent_req *subreq);
+
+struct tevent_req *
+sss_dp_resolver_get_send(TALLOC_CTX *mem_ctx,
+                         struct resp_ctx *rctx,
+                         struct sss_domain_info *dom,
+                         bool fast_reply,
+                         uint32_t entry_type,
+                         const char *name,
+                         const char *address)
+{
+    struct sss_dp_resolver_get_state *state;
+    struct tevent_req *req;
+    struct tevent_req *subreq;
+    struct be_conn *be_conn;
+    uint32_t dp_flags;
+    errno_t ret;
+
+    req = tevent_req_create(mem_ctx, &state, struct sss_dp_resolver_get_state);
+    if (req == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Unable to create tevent request!\n");
+        return NULL;
+    }
+
+    /* either, or, not both */
+    if (name != NULL && address != 0) {
```

Comment:
Since `address` is a char pointer as well please use `address != NULL`.

The case where both are NULL is not handled here, is this expected?

Since both parameter are of the same type, did you consider to use a single char pointer for the value and a bool or enum to indicate if it is a name or an address? I know that this style is used for other requests as well, but typically in those requests the parameters are of different types (char pointer and some integer type). You do not have to change this I'm just wondering if you tried to be similar to other request or if there might be even benefits doing it this way.